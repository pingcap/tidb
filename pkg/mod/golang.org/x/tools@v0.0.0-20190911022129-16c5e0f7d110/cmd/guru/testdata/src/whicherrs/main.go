package main

type errType string

const constErr errType = "blah"

func (et errType) Error() string {
	return string(et)
}

var errVar error = errType("foo")

func genErr(i int) error {
	switch i {
	case 0:
		return constErr
	case 1:
		return errVar
	default:
		return nil
	}
}

func unreachable() {
	err := errVar // @whicherrs func-dead "err"
	_ = err
}

func main() {
	err := genErr(0) // @whicherrs localerrs "err"
	_ = err
}
