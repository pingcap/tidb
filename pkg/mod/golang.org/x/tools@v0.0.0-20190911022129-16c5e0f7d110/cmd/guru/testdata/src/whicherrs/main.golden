-------- @whicherrs func-dead --------

Error: pointer analysis did not find expression (dead code?)
-------- @whicherrs localerrs --------
this error may point to these globals:
	errVar
this error may contain these constants:
	constErr
this error may contain these dynamic types:
	errType

